nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RetryCount.java

```diff
@@ -0,0 +1,210 @@
+            .name("Input queue data size limit")
+            .displayName("Input queue data size limit")
+            .description("If the total input queue data size (a combination of all connections feeding this processor) is above this configured limit then any processed FlowFile is routed to " +
+                    "'over limit'. This is necessary to avoid a back-pressure deadlock caused by with circular back-pressure. The default values are set to be half of what the default " +
+                    "configuration values are for connections back-pressure.")
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .expressionLanguageSupported(VARIABLE_REGISTRY)
+            .defaultValue("500 mb")
+            .build();
+
+    public static final PropertyDescriptor PROP_INPUT_QUEUE_COUNT_LIMIT = new PropertyDescriptor.Builder()
+            .name("Input queue count limit")
+            .displayName("Input queue count limit")
+            .description("If the total input queue FlowFile count (a combination of all connections feeding this processor) is above this configured limit then any processed FlowFile is routed to " +
+                    "'over limit'. This is necessary to avoid a back-pressure deadlock caused by with circular back-pressure. The default values are set to be half of what the default " +
+                    "configuration values are for connections back-pressure.")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(VARIABLE_REGISTRY)
+            .defaultValue("5000")
+            .build();
+
+
+    public static final Relationship REL_RETRY = new Relationship.Builder()
+            .name("retry")
+            .description("FlowFiles that have not yet exceeded the retry limit are routed here.")
+            .build();
+
+    public static final Relationship REL_OVER_LIMIT = new Relationship.Builder()
+            .name("over limit")
+            .description("A FlowFile which has exceeded the limit for the number of retries is routed here.")
+            .build();
+
+
+    private static final List<PropertyDescriptor> descriptors;
+    private static final Set<Relationship> relationships;
+
+    static {
+        final List<PropertyDescriptor> descriptorsTemp = new ArrayList<PropertyDescriptor>();
+        descriptorsTemp.add(PROP_RETRY_LIMIT);
+        descriptorsTemp.add(PROP_PENALIZE_FLOWFILE);
+        descriptorsTemp.add(PROP_WARN_ON_OVER_LIMIT);
+        descriptorsTemp.add(PROP_INPUT_QUEUE_DATA_SIZE_LIMIT);
+        descriptorsTemp.add(PROP_INPUT_QUEUE_COUNT_LIMIT);
+        descriptors = Collections.unmodifiableList(descriptorsTemp);
+
+        final Set<Relationship> relationshipsTemp = new HashSet<Relationship>();
+        relationshipsTemp.add(REL_OVER_LIMIT);
+        relationshipsTemp.add(REL_RETRY);
+        relationships = relationshipsTemp;
```

Comment:
You have it for property descriptors, probably should make relationships an unmodifiable set also:
`relationships = Collections.unmodifiableSet(relationshipsTemp);`